android/src/main/java/com/github/yamill/orientation/listeners/OrientationListener.kt

```diff
@@ -0,0 +1,67 @@
+package com.github.yamill.orientation.listeners
+
+import android.app.Activity
+import android.util.Log
+import android.view.OrientationEventListener
+import com.facebook.common.logging.FLog
+import com.facebook.react.bridge.Arguments
+import com.facebook.react.modules.core.DeviceEventManagerModule
+import com.facebook.react.bridge.LifecycleEventListener
+import com.facebook.react.bridge.ReactContext
+import com.facebook.react.common.ReactConstants
+import com.github.yamill.orientation.throttleLatest
+import kotlinx.coroutines.GlobalScope
+
+class OrientationListener internal constructor(
+        private val reactContext: ReactContext,
+        private val onGetCurrentActivity: () -> Activity?,
+) : LifecycleEventListener {
+
+    private lateinit var orientationEventListener: OrientationEventListener;
+
+    override fun onHostResume() {
+        val activity = onGetCurrentActivity()
+        if (activity != null && !this::orientationEventListener.isInitialized) {
+            orientationEventListener = object: OrientationEventListener(activity) {
+                override fun onOrientationChanged(orientationDegrees: Int) {
+                    onOrientationDegreesChange(orientationDegrees, reactContext)
+                }
+            }
+
+            if (orientationEventListener.canDetectOrientation()) {
+                orientationEventListener.enable();
+            }
+        } else {
+            FLog.e(ReactConstants.TAG, "no activity to register receiver")
+        }
+
+    }
+
+    override fun onHostPause() = Unit
+    override fun onHostDestroy() = Unit
+
+    companion object {
+        var lastEmittedOrientationDegrees: Int? = null
+
+        val onOrientationDegreesChange: (Int, ReactContext) -> Unit = throttleLatest(
+                intervalMs = 1000L,
+                coroutineScope = GlobalScope,
+                ::tryEmitOrientationDegreesChange
+        )
+
+        private fun tryEmitOrientationDegreesChange(orientationDegrees: Int, reactContext: ReactContext) {
+            if (reactContext.hasActiveReactInstance()) {
+                val params = Arguments.createMap()
+
+                if (orientationDegrees != lastEmittedOrientationDegrees) {
```

Comment:
If we wanted to here, could we instead only emit if difference between `orientationDegrees` and `lastEmittedOrientationDegrees` greater than > x?

Not sure if that'd be useful/helpful but wanted to poke more at the de-spamming idea

Reply:
In practice, without this distinct-until-changed logic, I saw the device emit events like this when I was holding the phone in my hand:
- 359
- 359
- 359
- 358
- 358
- 359
- 0

I'm okay with this emitting 359 and then 358, and I was mostly trying to get rid of the sequence of 359s. I'm not sure why this callback emitted this so many times. Maybe the sensor was capturing fractional degrees but the callback rounds to an integer?